crates/transcode/src/env_check.rs

```diff
@@ -0,0 +1,89 @@
+use scale_info::{
+    form::PortableForm,
+    Field,
+    PortableRegistry,
+    TypeDef,
+};
+
+use anyhow::{
+    Context,
+    Result,
+};
+
+use crate::ContractMessageTranscoder;
+
+#[derive(Debug, Clone)]
+pub enum EnvCheckError {
+    CheckFailed(String),
+    RegistryError(String),
+}
+
+impl From<anyhow::Error> for EnvCheckError {
+    fn from(value: anyhow::Error) -> Self {
+        Self::RegistryError(value.to_string())
+    }
+}
+
+fn get_node_env_fields(registry: &PortableRegistry) -> Result<Vec<Field<PortableForm>>> {
+    let env_type = registry
+        .types
+        .iter()
+        .find(|t| t.ty.path.segments == ["pallet_contracts", "Environment"])
+        .context("The node does not contain `Environment` type. Are you using correct `pallet-contracts` version?")?;
+
+    if let TypeDef::Composite(composite) = &env_type.ty.type_def {
+        Ok(composite.fields.clone())
+    } else {
+        anyhow::bail!("`Environment` type definition is in the wrong format");
+    }
+}
+
+pub fn resolve_type_definition(
+    registry: &PortableRegistry,
+    id: u32,
+) -> Result<TypeDef<PortableForm>> {
+    let tt = registry
+        .resolve(id)
+        .context("Type is not present in registry")?;
+    if tt.type_params.is_empty() {
+        if let TypeDef::Composite(comp) = &tt.type_def {
+            println!("Resolve type definition: {:#?}", tt);
+            let tt_id = comp
+                .fields
+                .get(0)
```

Comment:
We should check for invalid composites which have more than one field. Maybe a test for that?